From f55fc5bb3f5d6aa634b120266b65d26a9d47456a Mon Sep 17 00:00:00 2001
From: Dennis Murimi <d.murimi14@gmail.com>
Date: Wed, 31 Jul 2024 16:06:53 +0300
Subject: [PATCH 1/3] ImsService: Patch to work with shared mtk libs

        <uses-library android:name="mediatek-carrier-config-manager" android:required="true"/>
        <uses-library android:name="mediatek-common" android:required="true"/>
        <uses-library android:name="mediatek-framework" android:required="true"/>
        <uses-library android:name="mediatek-ims-base" android:required="true"/>
        <uses-library android:name="mediatek-ims-common" android:required="true"/>
        <uses-library android:name="mediatek-telephony-common" android:required="true"/>
        <uses-library android:name="mediatek-telecom-common" android:required="true"/>
        <uses-library android:name="mediatek-telephony-base" android:required="true"/>

Signed-off-by: Dennis Murimi <d.murimi14@gmail.com>
---
 AndroidManifest.xml | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index a0c5f7b..889dbe1 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -54,6 +54,14 @@
     <application android:allowBackup="false" android:appComponentFactory="androidx.core.app.CoreComponentFactory" android:directBootAware="true" android:extractNativeLibs="false" android:label="ImsApp" android:name="com.mediatek.ims.ImsApp" android:persistent="true" android:usesNonSdkApi="true">
         <uses-library android:name="com.mediatek.ims.oemplugin" android:required="false"/>
         <uses-library android:name="com.mediatek.ims.plugin" android:required="false"/>
+        <uses-library android:name="mediatek-carrier-config-manager" android:required="true"/>
+        <uses-library android:name="mediatek-common" android:required="true"/>
+        <uses-library android:name="mediatek-framework" android:required="true"/>
+        <uses-library android:name="mediatek-ims-base" android:required="true"/>
+        <uses-library android:name="mediatek-ims-common" android:required="true"/>
+        <uses-library android:name="mediatek-telephony-common" android:required="true"/>
+        <uses-library android:name="mediatek-telecom-common" android:required="true"/>
+        <uses-library android:name="mediatek-telephony-base" android:required="true"/>
         <service android:exported="true" android:name="com.mediatek.ims.MtkDynamicImsService" android:permission="android.permission.BIND_IMS_SERVICE">
             <meta-data android:name="android.telephony.ims.RCS_FEATURE" android:value="true"/>
             <meta-data android:name="android.telephony.ims.MMTEL_FEATURE" android:value="true"/>
-- 
2.39.5

